newIDE/app/src/EventsSheet/EventsTree/Instruction.js

```diff
@@ -146,14 +166,44 @@ const Instruction = (props: Props) => {
           }
 
           const parameterMetadata = metadata.getParameter(parameterIndex);
-          const parameterType = parameterMetadata.getType();
+          const parameterSubType = parameterMetadata.getType();
+          const parameterType = gd.ParameterMetadata.isExpression(
+            'number',
+            parameterSubType
+          )
+            ? 'number'
+            : gd.ParameterMetadata.isExpression('string', parameterSubType)
+            ? 'string'
+            : parameterSubType;
```

Comment:
The parser should probably understand the subtypes of "string" and "number" when they are passed in `ParseExpression()` as it already does these check in some internal steps.

Reply:
Probably yes!